### ISSUE TITLE
Consider renaming `build` to `compile`?

### ISSUE THREAD

**Issue by garyb** [purescript-contrib/pulp#326]
This would better match the underlying `purs` command, but primarily I think it would be better as `build` doesn't do really do what you'd normally consider building: it does't produce a usable final asset. I guess that's slightly arguable, as if you're using `node` then it kinda does, but for all other purposes that's not true.

I meant to write this up a while ago, after I saw a few people asking how to actually build a PS project for the web.

If we were to do this I'd suggest keeping the `build` command too and printing a warning that it is being deprecated in favour of `compile`, and then eventually drop it.

**Comment by hdgarrood** [comment]
I'm not a fan, really:
- `build` is very often used together with the `--to` option, which (in most cases) does produce a usable final asset,
- Besides, I don't think there's any established consensus that "building" means producing a usable final asset. For instance, I use `stack build` when I'm working on a library in Haskell. Also, with interpreted languages, since there's no compile step, the "CI build" usually just means running the tests and seeing if they still pass. Systems like Jenkins have diluted the term "build" even further.
- there's already a lot of docs around using `pulp build`, and this isn't really a strong enough reason to me to require them all to be updated.

**Comment by garyb** [comment]
- `build --to` is a fair point, but maybe that should actually be a separate command rather than a flag
- `stack build` does produce finally built targets though, it just doesn't do anything with them
- that's why I suggested a deprecation message :wink:

Just a general point: at the moment our response to changing stuff is often "well, it's been fine this way up until now" but I think if we seriously want to address issues newcomers have to the language and tools we do have to be a bit more open minded about these things. This isn't meant as a criticism of you, I'm equally guilty of it - I'm also not suggesting we back down when it's issues of correctness or the like, just more things where it's about the UX.

**Comment by garyb** [comment]
I'm also not saying _we must do this!_ :wink: it was just an idea.

**Comment by hdgarrood** [comment]
A deprecation message certainly mitigates the issue, but I think it's still poor UX to run into one at all. Also, even though `stack build` does produce finally built targets when working on a library (good point, my mistake), you barely ever actually use them directly; instead you depend on those libraries in other projects.

I do seriously want to address issues newcomers have with the language and tools, but I don't think the argument I've put forward here is closed-minded. I just think that doing this would cause more confusion than it would prevent, especially amongst people who are still getting to grips with `pulp`.

**Comment by garyb** [comment]
> I do seriously want to address issues newcomers have with the language and tools, but I don't think the argument I've put forward here is closed-minded. I just think that doing this would cause more confusion than it would prevent, especially amongst people who are still getting to grips with `pulp`.

No, that's fair enough - it was just something I thought of in relation to the last point about there being stuff out there already about/using `pulp build`, that we shouldn't perhaps weight a point like that too heavily where there are UX issues.

This particular issue isn't something I've ever even though about until I saw some people confused by it (also `n=3` or something, which is why I don't feel all that strongly about it). It's just been on my mind as I recently spent I think maybe 30 minutes trying to figure out how to use `ghcid` on the compiler project before just asking Christoph, as he'd previously told me the magic arguments needed to make it work. It was pretty frustrating!
